compiler/rustc_resolve/src/lib.rs

```diff
@@ -648,7 +648,7 @@ impl<'ra> Module<'ra> {
     {
         for (key, name_resolution) in resolver.as_mut().resolutions(self).borrow().iter() {
             if let Some(binding) = name_resolution.borrow().best_binding() {
-                f(resolver, key.ident, key.ns, binding);
+                f(resolver, key.ident.0, key.ns, binding);
```

Comment:
```suggestion
                f(resolver, key.ident, key.ns, binding);
```
If we already have a `ModernIdent` on hands then we should preserve it for the callback.